src/test/java/plugins/GitLabPluginTest.java

```diff
@@ -49,307 +64,559 @@ public class GitLabPluginTest extends AbstractJUnitTest {
+                        } catch (Exception ex) {
+                            return null;
+                        }
+                    }
+                });
+    }
+
+    private Group createGroupViaApi(GroupApi groupApi, String groupName, String groupPath) {
+        var groupParams =
+                new GroupParams().withName(groupName).withPath(groupPath).withMembershipLock(false);
+        return waitFor()
+                .withMessage("Creating GitLab group '%s'", groupName)
+                .withTimeout(GITLAB_API_RETRY_TIMEOUT)
+                .until(() -> {
+                    try {
+                        return groupApi.createGroup(groupParams).withVisibility(Visibility.PRIVATE);
+                    } catch (Exception e) {
+                        try {
+                            return groupApi.getGroup(groupPath);
+                        } catch (Exception ex) {
+                            return null;
+                        }
+                    }
+                });
+    }
+
+    private Member addGroupMemberViaApi(GroupApi groupApi, Long groupId, Long userId, AccessLevel accessLevel) {
+        return waitFor()
+                .withMessage("Adding user %d to group %d", userId, groupId)
+                .withTimeout(GITLAB_API_RETRY_TIMEOUT)
+                .until(() -> {
+                    try {
+                        return groupApi.addMember(groupId, userId, accessLevel);
+                    } catch (Exception e) {
+                        try {
+                            return groupApi.getMember(groupId, userId);
+                        } catch (Exception ex) {
+                            return null;
+                        }
+                    }
+                });
+    }
+
+    private MergeRequest createMergeRequestViaApi(
+            GitLabApi gitlabapi, Project project, String sourceBranch, String targetBranch, String mrTitle) {
+        var params = new MergeRequestParams()
+                .withSourceBranch(sourceBranch)
+                .withTargetBranch(targetBranch)
+                .withTitle(mrTitle);
+        return waitFor()
+                .withMessage("Creating merge request '%s'", mrTitle)
+                .withTimeout(GITLAB_API_RETRY_TIMEOUT)
+                .until(() -> {
+                    try {
+                        return gitlabapi.getMergeRequestApi().createMergeRequest(project, params);
+                    } catch (Exception e) {
+                        try {
+                            var filter = new MergeRequestFilter().withProjectId(project.getId());
+                            return gitlabapi.getMergeRequestApi().getMergeRequests(filter).stream()
+                                    .filter(mr -> mrTitle.equals(mr.getTitle()))
+                                    .findFirst()
+                                    .orElse(null);
+                        } catch (Exception ex) {
+                            return null;
+                        }
```

Comment:
add .ignoring(Exception) to the wait (or better still be specific about the exception?) and reduce this block indentation.

otherwise this swallows exceptions in the case of a timeout.